src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2397,8 +2468,20 @@ else if (4 >= bulkScale)
                             tdsWriter.writeByte((byte) 0x04);
                         else
                             tdsWriter.writeByte((byte) 0x05);
-
-                        tdsWriter.writeTime((java.sql.Timestamp) colValue, bulkScale);
+                        if (colValue instanceof String) {
```

Comment:
Can you explain this with code comments? Also why do we have to parse for TIME, not for other types?

Reply:
I added comments. Other datatypes are already handled as if the data was string (this is the case for date, datetime and smalldatetime), so there was no need for extra steps. Time was the only datatype (other than DTO) that the code was assuming was coming on the form of a timestamp, so I needed to adjust accordingly.